src/python/pants/option/config_test.py

```diff
@@ -1,24 +1,134 @@
 # Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
 # Licensed under the Apache License, Version 2.0 (see LICENSE).
 
+import configparser
 from textwrap import dedent
-from typing import Dict
 
+import pytest
 from twitter.common.collections import OrderedSet
 
 from pants.option.config import Config
 from pants.testutil.test_base import TestBase
 from pants.util.contextutil import temporary_file
 
 
-class ConfigTest(TestBase):
+FILE1_INI = dedent(
+  """
+  [DEFAULT]
+  name: foo
+  answer: 42
+  scale: 1.2
+  path: /a/b/%(answer)s
+  embed: %(path)s::foo
+  disclaimer:
+    Let it be known
+    that.
 
-  def _setup_config(self, config1_content: str, config2_content: str, *, suffix: str) -> Config:
-    with temporary_file(binary_mode=False, suffix=suffix) as config1, \
-      temporary_file(binary_mode=False, suffix=suffix) as config2:
-      config1.write(config1_content)
+  [a]
+  list: [1, 2, 3, %(answer)s]
+  list2: +[7, 8, 9]
+  list3: -["x", "y", "z"]
+  list4: +[0, 1],-[8, 9]
```

Comment:
This is new. It's meant to test that we correctly handle both appending and filtering to a list option at the same time.